python/mxnet/optimizer.py

```diff
@@ -143,8 +143,8 @@ def _update_count(self, index):
         """
         if index not in self._index_update_count:
             self._index_update_count[index] = self.begin_num_update
+        self.num_update = self._index_update_count[index]
```

Comment:
@mli Why was the max introduced in the first place? I remember it has something to do with distributed training?